covid19-dashboard/dc-covid19/src/Panel.tsx

```diff
@@ -15,48 +15,45 @@
  */
 
 import React from "react";
-import {absolute, difference, perCapita, increase} from "./DataCalculator";
+import {calculate} from "./DataCalculator";
 import ContentFile from "./ContentFile.json";
 import EmptyPanel from "./EmptyPanel";
-import BarGraph from "./BarGraph";
 import {getRangeOfDates} from "./Utils";
+import Graph from "./Graph";
+import generateGraphMetadata from "./MetaDataGenerator";
 
 type Props = {
-    data: any[],
+    allData: any[],
     label: string,
     region: string,
     datePicked: string,
     selectedShowTopN: number,
     panelId: string
 }
-type DataPerGeoIdPerDate = {string: {string: number}} | {}
+type DateToGeoIdToValue = {date: {geoId: number}} | {}
 
 export default function Panel(props: Props) {
-    let data: DataPerGeoIdPerDate = {}
     const content = ContentFile[props.panelId]
     const deltaDays: number = content.deltaDays
 
     // Is it cases or deaths?
-    const inputData: DataPerGeoIdPerDate = props.label === 'cases' ? props.data[0] : props.data[1]
-    const geoIdToPopulation: {string: number} = props.data[2]
+    const inputData: DateToGeoIdToValue = props.label === 'cases' ? props.allData[0] : props.allData[1]
+    const geoIdToPopulation: {geoId: number} = props.allData[2]
+    const geoIdToName: {geoId: string} = props.allData[3]
     const rangeOfDates: [string, string] = getRangeOfDates(props.datePicked, deltaDays)
 
-    // What type of data do we want to calculate? Check the type on the content file
-    switch (content.type){
-        case 'absolute':
-            data = absolute(inputData, rangeOfDates)
-            break;
-        case 'difference':
-            data = difference(inputData, rangeOfDates, deltaDays)
-            break;
-        case 'perCapita':
-            data = perCapita(inputData, rangeOfDates, deltaDays, geoIdToPopulation)
-            break;
-        case 'increase':
-            data = increase(inputData, rangeOfDates, deltaDays)
-            break;
+    // Calculate per-capita.
+    const calculatedData: DateToGeoIdToValue = calculate(inputData, rangeOfDates, deltaDays, content.dataType, geoIdToPopulation)
+
+    let absoluteIncrease: DateToGeoIdToValue = {}
+    if (content.dataType=== 'increase') {
+        absoluteIncrease = calculate(inputData, rangeOfDates, deltaDays, content.dataType, geoIdToPopulation);
```

Comment:
**Instead of content.dataType, it should be "difference" as an parameter string. I will fix that**

Guha wants to show the absolute increase number on-hover (for the percent increase graphs). So we also have to calculate the absolute increase. That's why there is an if-statement checking to see if dataType is 'increase'.